new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDomainInPV.java

```diff
@@ -295,38 +308,111 @@ public void testDomainInPvUsingWlst() {
+                .clusterName(wdtClusterName)
+                .replicas(wdtReplicaCount)
                 .serverStartState("RUNNING")));
 
     // verify the domain custom resource is created
-    createDomainAndVerify(domain, wlstDomainNamespace);
+    createDomainAndVerify(domain, wdtDomainNamespace);
 
     // verify admin server pod is ready
-    checkPodReady(adminServerPodName, domainUid, wlstDomainNamespace);
+    checkPodReady(wdtAdminServerPodName, wdtDomainUid, wdtDomainNamespace);
 
     // verify the admin server service created
-    checkServiceExists(adminServerPodName, wlstDomainNamespace);
+    checkServiceExists(wdtAdminServerPodName, wdtDomainNamespace);
 
     // verify managed server pods are ready
-    for (int i = 1; i <= replicaCount; i++) {
+    for (int i = 1; i <= wdtReplicaCount; i++) {
       logger.info("Waiting for managed server pod {0} to be ready in namespace {1}",
-          managedServerPodNamePrefix + i, wlstDomainNamespace);
-      checkPodReady(managedServerPodNamePrefix + i, domainUid, wlstDomainNamespace);
+          wdtManagedServerPodNamePrefix + i, wdtDomainNamespace);
+      checkPodReady(wdtManagedServerPodNamePrefix + i, wdtDomainUid, wdtDomainNamespace);
     }
 
     // verify managed server services created
-    for (int i = 1; i <= replicaCount; i++) {
+    for (int i = 1; i <= wdtReplicaCount; i++) {
       logger.info("Checking managed server service {0} is created in namespace {1}",
-          managedServerPodNamePrefix + i, wlstDomainNamespace);
-      checkServiceExists(managedServerPodNamePrefix + i, wlstDomainNamespace);
+          wdtManagedServerPodNamePrefix + i, wdtDomainNamespace);
+      checkServiceExists(wdtManagedServerPodNamePrefix + i, wdtDomainNamespace);
     }
+    previousTestSuccessful = true;
+  }
+
+  /**
+   * Deploy an application using admin t3 channel port, target to both cluster and admin server.
+   */
+  @Test
+  @Order(2)
+  @DisplayName("Deploy an application using t3 channel port")
+  public void testDeployAppUsingT3ChannelPort() {
+    Assumptions.assumeTrue(previousTestSuccessful);
+    previousTestSuccessful = false;
+    logger.info("Getting node port for T3 channel");
+    int t3channelNodePort = assertDoesNotThrow(() -> getServiceNodePort(
+        wdtDomainNamespace, wdtAdminServerPodName + "-external", "t3channel"),
+        "Getting admin server t3channel node port failed");
+    assertNotEquals(-1, wdtT3ChannelPort, "admin server t3channelport is not valid");
+
+    //deploy application using t3 channel port
+    Path archivePath = Paths.get(ITTESTS_DIR, "../src/integration-tests/apps/testwebapp.war");
+    logger.info("Deploying webapp {0} to domain using t3 channelport", archivePath);
+    DeployUtil.deployUsingWlst(K8S_NODEPORT_HOST, Integer.toString(t3channelNodePort),
+        ADMIN_USERNAME_DEFAULT, ADMIN_PASSWORD_DEFAULT, wdtClusterName + "," + wdtAdminServerName, archivePath,
+        wdtDomainNamespace);
+
+    previousTestSuccessful = true;
```

Comment:
If the tests have dependencies or ordered tests, with out this logic, it will run the other tests even if the first test fails and they take time to execute and eventually fail.

Reply:
Yes that is one of the reason I am hesitant to move the deploy assertions out of test method.

Reply:
not just in this test, this applies for all tests where there are dependencies, we have such tests in new/JUnit5 tests. Why do we want execute a test when its prerequisites have not met?

Reply:
The prerequisites  like domain creation and startup are asserted, so if prerequisites are not met this is not going to get executed.

Reply:
I mean if test1(lets say domain creation) is prerequisite to test2(testing some feature like load balancing in this test) then we don't want to run test2 if test1 fails. In such cases, we need this logic.